### ISSUE TITLE
Uhuuull 

### ISSUE THREAD

**Issue by AdamTeodoro** [angular/angularfire#3043]
A ferramenta muda a maneira de ser utilizada todos os dias assim meu projeto sempre fica desatualizado, está na hora de mudar as mudanças?
tenho mais de 1000 dependências se cada uma delas decidirem mudarem repentinamente não compensa mais programar em cima dessas dependências.

**Comment by Sapython** [comment]
Please try to write your issue in English. And follow the issue template.

**Comment by AdamTeodoro** [comment]
Be careful with thethe culture of update your tools and ignore the users that are using !, this generate the big impact in the cummunity of devs, and create the big transturns and vulnerablities caused by the abandoned of the tecnology, the peopleas dont update your code to adequaly your sistems due the "breaking changes".
This update generate the most discredit, example: js is js and have the problems.
angularfire is angularfire and have problems.
the JS not destroy your sintaxe to solve the problems. angularjs destroy your sintexe to apply tree shake tecnices if this solve the problema,  this also cause others problems

**Comment by jamesdaniels** [comment]
Closing as off topic & unproductive.

Feel free to use discussions for conversations, but honestly breaking after 2 years of relative stability to offer tree-shaking to developers is worth the cost IMO. 

I think it's fair to say that the JS ecosystem (by large) should be more mindful of breaking changes, maintaining libraries, and shuffling folk around; but we worked our butt off to keep this library stable. We did our best to be mindful of semver and have shipped and are committed to maintaining full compatibility with the old API under `/compat` for the time being.
